core/src/main/java/com/rudderstack/android/sdk/core/RudderMessage.java

```diff
@@ -47,7 +47,8 @@
 
     RudderMessage() {
         context = RudderElementCache.getCachedContext();
-        this.anonymousId = context.getDeviceId();
+        RudderPreferenceManager preferenceManger = RudderPreferenceManager.getInstance(RudderClient.getInstance().getApplication());
```

Comment:
Completely personal opinion: Can we introduce a method like `RudderPreferenceManager.getInstance()` and take the application context within that if needed? 

Passing the `RudderClient.getInstance().getApplication()` everytime looks unnecessary to me because we are any way taking it form a static context.

Reply:
chnaged this